components/tools/OmeroJava/test/integration/gateway/TablesFacilityTest.java

```diff
@@ -276,5 +281,46 @@ public void testThreshold() throws Exception {
         Assert.assertEquals(td.getData()[0].length,
                 TablesFacility.DEFAULT_MAX_ROWS_TO_FETCH);
     }
+    
+    @Test(dependsOnMethods = { "testThreshold" })
+    public void testUpdateTable() throws Exception {
+        // modify values for row 10 to 20, columns 5, 6 and 7
+        TableData td = tablesFacility.getTable(rootCtx,
+                original.getOriginalFileId(), 10, 20, new int[] { 5, 6, 7 });
 
+        for (int c = 0; c < td.getColumns().length; c++) {
+            Class<?> type = td.getColumns()[c].getType();
+            for (int r = 0; r < td.getData()[0].length; r++) {
+                if (type.equals(String.class)) {
+                    td.getData()[c][r] = "newValue";
```

Comment:
Not part of the scope for this PR, but getData method is used for setting data here (which could be confusing for the consumers of the api). I would have expected something of the order of:
 `newValue = td.getData()[c][r]; 
   td.setData('newValue');`